openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/BallerinaSchemaGenerator.java

```diff
@@ -356,7 +356,7 @@ private RecordTypeDescriptorNode getAllOfRecordTypeDescriptorNode(OpenAPI openAp
             if (allOfSchema.getType() == null && allOfSchema.get$ref() != null) {
                 //Generate typeReferenceNode
                 getAllOfRecordFieldForReference(schemaDoc, recordFieldList, allOfSchema);
-            } else if (allOfSchema instanceof ObjectSchema && (allOfSchema.getProperties() != null)) {
+            } else if (allOfSchema instanceof ObjectSchema || allOfSchema.getProperties() != null) {
```

Comment:
Won't this `||`  condition be lead to an NPE?

Reply:
yes there is a possibility of getting a NPE if user has defined any Object schema without any properties. 
```
allOf:
        - description: ""
          type: object
        - properties:
            entries:
              type: array
              items:
                type: string
```